src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataResourceDeserializer.cs

```diff
@@ -226,20 +247,23 @@ public virtual object CreateResourceInstance(IEdmStructuredTypeReference structu
 
                 if (readContext.IsDeltaOfT)
                 {
-                    IEnumerable<string> structuralProperties = structuredType.StructuralProperties()
+                    IEnumerable<string> structuralProperties = structuredType.StructuredDefinition().Properties()
```

Comment:
Why this change? Does `structuredType.StructuredDefinition().Properties()` return something different from `structuredType.StructuralProperties()`? Does it for example return navigation properties as well? Because if it does, we should rename the `structuralProperties` variable...

Reply:
@gathogojr yes this returns `structuredType.StructuredDefinition().Properties()` this returns even navigation properties the other one doesn't.

Reply:
what did you have in mind on renaming it?

Reply:
@ElizabethOkerio What I meant is, if you have a variable named `structuralProperties` and it contains both structured and navigation properties, isn't that misleading? I think you should drop `structural` and just name it `properties`